Remove more VMThread reclamation logic

0xdaryl opened the pull request:
* Remove calls to setVMThreadRequired from x86
* Remove OMR::Symbol isVMThreadLive flag
* Remove vmThreadLiveCount from OMR::CodeGenerator
* Remove obsolete OMR::X86::Instruction RestoreVMThreadInstruction
* Remove deferred edge splitting vmThread optimization from x86
* Remove VMThreadRestoringLabel field from LabelSymbol
* Remove specialty VMThread assignment logic from x86 register assigner

0xdaryl commented:
Notes to reviewers:

* Because testing using the FV tests in TravisCI does not exercise these code paths extensively, I ran an OpenJ9 build within IBM on Linux and Windows 32/64-bit platforms as well as Power and Z Linux using promotion-calibre tests and all tests were successful.
* I chose to stage this removal with multiple commits to make triage easier should OMR or downstream testing uncover problems relating to this in the future.

I would appreciate a review from @0dvictor please.

0dvictor commented:
LGTM.

fjeremic commented on compiler/codegen/OMRCodeGenerator.cpp:
```diff
@@ -977,27 +976,6 @@ OMR::CodeGenerator::canClobberNodesRegister(
 void
 OMR::CodeGenerator::setVMThreadRequired(bool v)
```
Will these be cleaned up in later PRs?

fjeremic commented on compiler/x/codegen/OMRCodeGenerator.cpp:
```diff
@@ -2702,16 +2701,6 @@ TR::Instruction *OMR::X86::CodeGenerator::splitBlockEntry(TR::Instruction *instr
-      if (label->getVMThreadRestoringLabel())
-         location = label->getVMThreadRestoringLabel()->getInstruction();
-      }
    location = location->getPrev();
```
Can this redefinition of `location` be fused up to the declaration? i.e.

```
TR::Instruction *location = instr->getPrev();
```

fjeremic commented on compiler/x/codegen/OMRX86Instruction.cpp:
```diff
@@ -4370,12 +4331,6 @@ generateLabelInstruction(TR_X86OpCodes     opCode,
-   {
-   return new (cg->trHeapMemory()) TR::X86RestoreVMThreadInstruction(RestoreVMThread, node, cg);
-   }
 
```
Can we remove the additional new lines? Similarly in a few other places in this PR.

0xdaryl commented:
@fjeremic : I rebased and pushed, which lost the original commits unfortunately.  Answers to your questions:

1)  Yes, those seemingly dead functions will be removed in a subsequent commit.  Some downstream project (OpenJ9) cleanup is needed first.

2)  I removed the creation and initialization of the `location` variable altogether and simply used `instr->getPrev()` instead.

3)  I'm not sure what spaces you're asking to be cleaned up.  Can you be more specific?

fjeremic commented on compiler/x/codegen/OMRX86Instruction.cpp:
```diff
@@ -434,14 +431,6 @@ TR::X86FenceInstruction::X86FenceInstruction(TR::Instruction   *precedingInstruc
-   : TR::Instruction(node, op, cg)
-   {
-   }
 
```
> I'm not sure what spaces you're asking to be cleaned up. Can you be more specific?

The double blank lines introduced by removal of these constructors. See line 433 and 434 after the change. There are several of these occurrences in this PR.

0xdaryl commented:
OMRX86Instruction.cpp is wildly inconsistent with its use of spaces between functions.  In the absence of any coding style guidelines for this project, I tend to keep style consistent with the local ecosystem unless its part of a larger refactoring effort.  Nevertheless, I did one sweep through OMRX86Instruction.cpp to impose a consistent one-space boundary between functions.  Rebased and pushed.

fjeremic commented:
@genie-omr build all
